src/Tools/dotnet-monitor/Egress/FileSystem/FileSystemEgressExtension.cs

```diff
@@ -195,5 +173,42 @@ private static string WrapMessage(string innerMessage)
                 return Strings.ErrorMessage_EgressFileFailedGeneric;
             }
         }
+
+        public Task<EgressArtifactResult> ValidateProviderAsync(string providerName,
+            EgressArtifactSettings settings,
+            CancellationToken token)
+        {
+            EgressArtifactResult result = new();
+            try
+            {
+                FileSystemEgressProviderOptions options = ValidateOptions(providerName);
+                result.Succeeded = true;
+            }
+            catch (Exception ex)
```

Comment:
Only catch OptionsValidationException; all other exceptions should flow through to help indicate some error in the system.